include/odp/api/spec/timer.h

```diff
@@ -158,6 +160,9 @@ typedef struct {
 	/** Clock source for timers */
 	odp_timer_clk_src_t clk_src;
 
+	/** Frequency of the constant external clock source.
+	 *  Only valid when clk_src is 'ODP_CLOCK_EXT_CONST_HZ'. */
+	uint64_t const_ext_hz;
```

Comment:
This could be simply clock frequency, which is by default zero (implementation knows the freq).

Reply:
If external frequency varies, then I believe passing clock frequency would not be helpful to the underlying timer hardware.

Reply:
My idea is that the enumeration is not changed above, but this additional parameter would tell if application knows the ext clock frequency to be fixed. CLOCK_EXT + non-zero clock_ext_hz value tells that the frequency is fixed. Value of zero means that application does not know the frequency, or it may vary.

Maybe clock_ext_hz would be better name for this than const_ext_hz.

Clock source enumeration may need to be extended later to cover multiple external (or internal) clock sources. We may then add more enums (e.g. ODP_CLOCK_EXT1, ODP_CLOCK_EXT2, ...) or another variable here (e.g. clock_ext_id).

Reply:
Makes sense, I removed the macro change in v4. Thanks.